dimod/constrained.py

```diff
@@ -652,7 +652,13 @@ def add_discrete_from_model(self,
             raise ValueError("discrete constraints must have at least two variables")
 
         for v, bias in qm.iter_linear():
-            if not vartype(v) is Vartype.BINARY:
+            if v in self.variables:
+                # it already exists, let's make sure it's not already used
+                if any(v in self.constraints[label].lhs.variables for label in self.discrete):
+                    raise ValueError(f"variable {v!r} is already used in a discrete variable")
```

Comment:
Not related to this PR, but why not something like this:
(f"variable {v.name} is already used in a discrete variable"
I know that v.name doesn't exist and need to do this in another way.

Reply:
Yeah, that's still an open feature request, see https://github.com/dwavesystems/dimod/issues/966